CoreScriptsRoot/Modules/VR/Panel3D.lua

```diff
@@ -0,0 +1,760 @@
+--Panel3D: 3D GUI panels for VR
+--written by 0xBAADF00D
+--revised/refactored 5/11/16
+
+local UserInputService = game:GetService("UserInputService")
+local RunService = game:GetService("RunService")
+local GuiService = game:GetService("GuiService")
+local CoreGui = game:GetService("CoreGui")
+local RobloxGui = CoreGui:WaitForChild("RobloxGui")
+
+
+--Util
+local Util = {}
+do
+	function Util.Create(instanceType)
+		return function(data)
+			local obj = Instance.new(instanceType)
+			for k, v in pairs(data) do
+				if type(k) == 'number' then
+					v.Parent = obj
+				else
+					obj[k] = v
+				end
+			end
+			return obj
+		end
+	end
+
+	-- RayPlaneIntersection (shortened)
+	-- http://www.siggraph.org/education/materials/HyperGraph/raytrace/rayplane_intersection.htm
+	function Util.RayPlaneIntersection(ray, planeNormal, pointOnPlane)
```

Comment:
Now I am seeing this function in many different files I am starting to think we should make a Math library module for this kind of code.